docs/resources/aws_kms_key.md

```diff
@@ -0,0 +1,175 @@
+      it { should exist }
+    end
+
+### Test that the specified key is enabled
+
+    describe aws_kms_key('arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321') do
+      it { should be_enabled }
+    end
+
+### Test that the specified key is rotation enabled
+
+    describe aws_kms_key('arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321') do
+      it { should be_rotation_enabled }
+    end
+
+<br>
+
+## Properties
+
+### key_id
+
+The globally unique identifier for the key.
+
+    describe aws_kms_key('arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321') do
+      its('key_id') { should cmp "id" }
+    end
+
+### arn
+
+The ARN identifier of the specified key. An ARN uniquely identifies the key within AWS.
+
+    describe aws_kms_key('arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321') do
+      its('arn') { should cmp "arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321" }
+    end
+
+### creation_date
+
+Specifies the date and time when the key was created.
+
+    describe aws_kms_key('arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321') do
+      its('creation_date') { should be < Time.now - 10 * 86400 }
+    end
+
+### created_days_ago
+
+Specifies the number of days since the key was created.
+
+    describe aws_kms_key('arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321') do
+      its('created_days_ago') { should be > 10 }
+    end
+
+### key_usage
+
+Specifies the cryptographic operations for which you can use the key. Currently the only allowed value is ENCRYPT_DECRYPT , which means you can use the key for the encrypt and decrypt operations.
+
+    describe aws_kms_key('arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321') do
+      its('key_usage') { should cmp "ENCRYPT_DECRYPT" }
+    end
+
+### key_state
+
+Specifies the state of the key one of "Enabled", "Disabled", "PendingDeletion", "PendingImport".
+
+    describe aws_kms_key('arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321') do
+      its('key_state') { should cmp "Enabled" }
+    end
+
+### description
+
+Specifies the description of the key.
+
+    describe aws_kms_key('arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321') do
+      its('description') { should cmp "key-description" }
+    end
+
+### deletion_date
+
+Specifies the date and time after which AWS KMS deletes the key. This value is present only when KeyState is PendingDeletion , otherwise this value is omitted.
+
+    describe aws_kms_key('arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321') do
+      its('deletion_date') { should cmp > Time.now + 7 * 86400 }
+    end
+
+### valid_to
+
+Provides the data and time until the key is not valid.  Once the key is not valid, AWS KMS deletes the key and it becomes unusable.  This value will be null unless the keys Origin is EXTERNAL and its ExpirationModel is KEY_MATERIAL_EXPIRES.
```

Comment:
This looks like AWS docs.  What is Null in Ruby-land?  Also, I don't think we're exposing ExpirationModel, so that may be confusing.